conman/routes/models.py

```diff
@@ -74,6 +84,64 @@ def handle(self, request, path):
         # Deal with the request
         return handler.handle(request, path)
 
+    def move_to(self, new_url, *, move_children):
+        """
+        Move this Route to a new url.
+
+        Requires `move_children` as a keyword argument. If True, all child
+        Routes will be moved along with this parent. Otherwise, when False, any
+        and all children will remain where they are.
+
+        In the event of a clash, an IntegrityError will be raised.
+        """
+        if move_children:
+            Route.objects.move_branch(self.url, new_url)
+            self.url = new_url
+        else:
+            old_url, self.url = self.url, new_url
+            try:
+                self.save()
+            except IntegrityError:
+                self.url = old_url
+                raise
```

Comment:
Come to think of it, `swap_with` is intended to be pretty safe. I can (just about) imagine how one might make it fail, but it's such an odd set of circumstances that I don't expect it to ever happen deliberately. Making it atomic seems wasteful, given any problem ought to be caught by `ATOMIC_REQUESTS`, and a 500 raised.